src/components/Timer.js

```diff
@@ -1,13 +1,70 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import CircularProgress from 'material-ui/CircularProgress';
-
+import { getTranslator } from '../reducers/translation';
 import styles from './Timer.scss';
 
 // TODO: this is just placeholder it needs to be implemented correctly
-export default () =>
-  (<div className={styles.timer}>
-    <CircularProgress size={25} />
-    <p className={styles.text}>
-      You have <span className={styles.countdown}>8 seconds</span> left
-    </p>
-  </div>);
+const TimerComponent = ({ i18n, secondsLeft }) =>
+  (secondsLeft >= 0 ?
+    <div className={styles.timer}>
+      <CircularProgress size={25} />
+      <p className={styles.text}>
+        {i18n('You have')} <span className={styles.countdown}>{secondsLeft} {i18n('seconds')}</span> {i18n('left.')}
+      </p>
+    </div>
+      :
+    <p>{i18n('Time is up you should start from scratch.')}</p>
+  );
+
+TimerComponent.propTypes = {
+  i18n: PropTypes.func.isRequired,
+  secondsLeft: PropTypes.number.isRequired,
+};
+
+const TimerComponentTranslated = connect(
+  state => ({ i18n: getTranslator(state) })
+  )(TimerComponent);
```

Comment:
You can also skip this connect and do both connections at the end (and pass `i18n` as a prop from `Timer` to `TiemrComponent`)